### PR TITLE
[JsonEncoder] Wire services

### PR DESCRIPTION
**Pull request by mtarld**
| Q             | A
| ------------- | ---
| Branch?       | 7.3
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Issues        | 
| License       | MIT

Follow-up of
* https://github.com/symfony/symfony/pull/51718

The FrameworkBundle part of the JsonEncoder component introduction.

The component related config is quite simple:
```yaml
framework:
    json_encoder:
        paths:
            App\EncodableDto\: '../src/EncodableDto/*'
```

Plus, the framework integration proposes the following bindings: 
- `EncoderInterface $jsonEncoder` to the `json_encoder.encoder` service 
- `DecoderInterface $jsonDecoder` to the `json_encoder.decoder` service

---

As this PR is based on top of https://github.com/symfony/symfony/pull/51718, only the last commit should be considered.

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -2542,4 +2544,38 @@ private function addHtmlSanitizerSection(ArrayNodeDefinition $rootNode, callable
...
+    private function addJsonEncoderSection(ArrayNodeDefinition $rootNode, callable $enableIfStandalone): void
+    {
+        $rootNode
+            ->children()
+                ->arrayNode('json_encoder')
+                    ->info('JSON encoder configuration')
+                    ->{$enableIfStandalone('symfony/json-encoder', EncoderInterface::class)}()
+                    ->fixXmlConfig('encodable_path')
+                    ->children()
+                        ->arrayNode('encodable_paths')
```
Comment (yceruto) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: Just `paths`? The info description already provides a good idea of what it is.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -2542,4 +2544,38 @@ private function addHtmlSanitizerSection(ArrayNodeDefinition $rootNode, callable
...
+                            ->info('Namespaces and paths of encodable/decodable classes.')
+                            ->normalizeKeys(false)
+                            ->useAttributeAsKey('namespace')
+                            ->scalarPrototype()->end()
+                            ->defaultValue([])
+                        ->end()
+                        ->arrayNode('cache_encoder_options')
```
Comment (yceruto) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: encoder_cache_options?

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -2542,4 +2544,38 @@ private function addHtmlSanitizerSection(ArrayNodeDefinition $rootNode, callable
...
+                            ->info('The options that will be used for encoder generation during cache warmup.')
+                            ->normalizeKeys(false)
+                            ->variablePrototype()->end()
+                            ->defaultValue([])
+                        ->end()
+                        ->arrayNode('cache_decoder_options')
```
Comment (yceruto) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: decoder_cache_options?

File: src/Symfony/Component/JsonEncoder/CacheWarmer/EncoderDecoderCacheWarmer.php [hunk 4]
```diff
@@ -37,12 +37,12 @@ final class EncoderDecoderCacheWarmer implements CacheWarmerInterface
...
-     * @param list<class-string>   $encodableClassNames
-     * @param array<string, mixed> $encoderOptions
-     * @param array<string, mixed> $decoderOptions
+     * @param iterable<class-string> $encodableClassNames
+     * @param array<string, mixed>   $encoderOptions
+     * @param array<string, mixed>   $decoderOptions
      */
     public function __construct(
-        private array $encodableClassNames,
+        private iterable $encodableClassNames,
```
Comment (OskarStark) [inline src/Symfony/Component/JsonEncoder/CacheWarmer/EncoderDecoderCacheWarmer.php]: This should be part of the initial PR, shouldn't it?

File: src/Symfony/Component/JsonEncoder/CacheWarmer/LazyGhostCacheWarmer.php [hunk 5]
```diff
@@ -25,10 +25,10 @@
 final class LazyGhostCacheWarmer extends CacheWarmer
 {
     /**
-     * @param list<class-string> $encodableClassNames
+     * @param iterable<class-string> $encodableClassNames
      */
     public function __construct(
-        private array $encodableClassNames,
+        private iterable $encodableClassNames,
```
Comment (OskarStark) [inline src/Symfony/Component/JsonEncoder/CacheWarmer/LazyGhostCacheWarmer.php]: This should be part of the initial PR, shouldn't it?

**Comment by chalasr** [comment]
@mtarld Can you please rebase this one?

**Comment by chalasr** [comment]
Thank you @mtarld.

**Review by chalasr — APPROVED** [review]

**Comment by chalasr** [comment]
Please don't forget to write some docs for this component. Thanks

**Pull request merged**